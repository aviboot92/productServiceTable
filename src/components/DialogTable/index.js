export {default} from './DialogTable.jsx';