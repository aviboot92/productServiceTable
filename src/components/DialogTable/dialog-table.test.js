import React from 'react';
import ReactDOM from 'react-dom';
import DialogTable from './DialogTable';
import ReactTestRenderer from 'react-test-renderer';

it('should render without exploding', () => {
    const div = document.createElement("div");
    const data = [
        {
            "academyId": 30021,
            "serialNumber": "TT-C67ML-A-0001-02583",
            "avgDailyDischarge": 0.2121
        }
    ];
    // It will be a failed test as data prop is missing which is compulsory for DialogTable at it should be in structure of data array as defined above
    ReactDOM.render(
        <DialogTable data={data}></DialogTable>, div)
});


it('matches snapshot', () => {
    const data = [
        {
            "academyId": 30015,
            "serialNumber": "TT-C67ML-A-0001-02522",
            "avgDailyDischarge": 1.0580
        }, {
            "academyId": 30015,
            "serialNumber": "1805C67HD02212",
            "avgDailyDischarge": 0.4624
        }, {
            "academyId": 30015,
            "serialNumber": "NGTT-0040",
            "avgDailyDischarge": 0.3634
        }, {
            "academyId": 30015,
            "serialNumber": "1805C67HD02244",
            "avgDailyDischarge": 0.3293
        }, {
            "academyId": 30015,
            "serialNumber": "1805C67HD02440",
            "avgDailyDischarge": 0.318
        }, {
            "academyId": 30015,
            "serialNumber": "TT-C67ML-A-0001-02573",
            "avgDailyDischarge": 0.3094
        }, {
            "academyId": 30015,
            "serialNumber": "TT-C67ML-A-0001-01355",
            "avgDailyDischarge": 0.2856
        }, {
            "academyId": 30015,
            "serialNumber": "1805C67HD01980",
            "avgDailyDischarge": 0.2129
        }
    ];
    const tree = ReactTestRenderer.create(<DialogTable data={data}></DialogTable>).toJSON();
    expect(tree).toMatchSnapshot();
})
