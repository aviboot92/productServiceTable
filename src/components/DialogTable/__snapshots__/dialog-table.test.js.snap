// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="dialogTable"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="customized table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-2 MuiTableCell-alignCenter"
            scope="col"
          >
            Avg (%)
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-2 MuiTableCell-alignCenter"
            scope="col"
          >
            Serial (No.)
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            106
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            TT-C67ML-A-0001-02522
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            46
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            1805C67HD02212
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            36
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            NGTT-0040
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            33
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            1805C67HD02244
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            32
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            1805C67HD02440
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            31
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            TT-C67ML-A-0001-02573
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            29
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            TT-C67ML-A-0001-01355
          </td>
        </tr>
        <tr
          className="MuiTableRow-root WithStyles(ForwardRef(TableRow))-root-4"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            21
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignCenter"
          >
            1805C67HD01980
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
